autoarray/fit/fit.py

```diff
@@ -41,7 +44,7 @@ def residual_map(self) -> abstract_structure.AbstractStructure:
         return fit_util.residual_map_from(data=self.data, model_data=self.model_data)
 
     @property
-    def normalized_residual_map(self) -> abstract_structure.AbstractStructure:
+    def normalized_residual_map(self) -> Union[np.ndarray, Array1D, Array2D]:

```

Comment:
If your Array1D and Array2D subclass np.ndarray there is no need for the Union - the most general type is np.ndarray.

It might be annoying that you can't use some attribute of Array1D without a lint error even though you know it's an Array1D. In that case you should be using type casting or using [generics](https://docs.python.org/3/library/typing.html#generics).